src/UFS2ARCO/ufsdataset.py

```diff
@@ -57,22 +58,22 @@ class UFSDataset:
 
     chunks_in = None
     chunks_out = None
-    coords = None
-    data_vars = None
-    zarr_name = None
+    coords = []
+    data_vars = []
```

Comment:
Since lists are mutable data types, it's actually not a good idea to set these as the default. The reason is that we could create an object, modify this attribute, then create another object, and it will have the modified attribute from the first object by default. I can explain this more if you like, it blew my mind when I first learned about it.

I also think having NoneType is cleaner for checking to see if the user provided coords or not in the yaml file later on (e.g. lines 109 and 185), but that's more subjective I think.

Reply:
This comes down to several things.  By defining the attribute as [], it has a known type of List.  None just lets my linter complain.  Checking the length of the list instead of for None can effectively tell you if anything is in the list.
This may be too long of a conversation for typing, but I don't follow the logic of not setting the type by default.  If you are concerned about deep vs. shallow copying, then just initialize the attributes in the __init__ method.

Reply:
Hmm. OK there's got to be a way to not run into this issue and appease the linter. I've always addressed it by making mutable class attributes default to None (like here). I'll look into it right now.

Reply:
This screenshot below shows an example of the issue I was talking about, which I was why I get nervous with mutable types as default arguments to class attributes. Since it is a little complicated I just avoid it by assigning None by default so I don't have to think too much, but I tested it out and your solution provided fixes the issue. 

![Screen Shot 2023-10-20 at 9 55 53 AM](https://github.com/HighTechnologyFoundation/UFS2ARCO/assets/28659578/41ec5485-950a-4ca4-a0ca-dce5aba452e6)

Even though bus2 is initialized as empty, it has a ghost rider on it! 👻